### PR TITLE
Verify build

### PR DESCRIPTION
**Pull request by tiainen**
## Description

Fixes pyrsia/pyrsia#818

This PR implements the `verify_transaction` function on the `VerificationService`. The implementation only works for builds that are reproducible, where the hashes of the built artifacts will always be the same.

It also introduces a `BuildEventClient` that is used by the different services to send build events on the build event loop.

## PR Checklist

- [x] I've read the [contributing guidelines](https://github.com/pyrsia/.github/blob/main/contributing.md).
- [x] I've read ["What is a Good PR?"](https://github.com/pyrsia/pyrsia/blob/main/docs/get_involved/good_pr.md)
- [x] I've included a good title and brief description along with how to review them.
- [x] I've linked any associated an [issue](https://github.com/pyrsia/pyrsia/issues).

### Code Contributions

- [x] I've built the code `cargo build --all-targets` successfully.
- [x] I've run the unit tests `cargo test --workspace` and everything passes.
- [x] I've made sure my rust toolchain is current `rustup update`.

### CODE REVIEW

**Comment by betarelease** [comment]
@efrisch and @fishseabowl added you as reviewers too

**Review by betarelease — APPROVED** [review]
The only thing that was not clear to me was how is verify_build being used and I found that the arch document delegates that to the consensus algorithm.

**Review by fishseabowl — APPROVED** [review]
LGTM

File: src/verification_service/service.rs [hunk 1]
```diff
@@ -14,66 +14,733 @@
...
     /// Verify a build for the specified transaction. This method is
     /// used to be able to reach consensus about a transaction that
     /// is a candidate to be committed to the blockchain.
     pub async fn verify_transaction(
-        &self,
-        _transaction: Transaction,
-        _sender: oneshot::Sender<Result<VerificationResult, VerificationError>>,
-    ) -> Result<(), VerificationError> {
+        &mut self,
+        transaction: &Transaction,
```
Comment (fishseabowl) [inline src/verification_service/service.rs:102]: I guess the verify service doesn't care about the transaction, only the payload of the transaction. So it's better to get the payload: &Vec<u8> as a parameter.

**Comment by mseabornIBM** [comment]
Could I get a link to the architecture document that explains all this?

**Comment by fishseabowl** [comment]
> Could I get a link to the architecture document that explains all this?

@mseabornIBM You can start by looking at https://github.com/pyrsia/pyrsia/blob/main/docs/developers/pyrsia-architecture-and-use-cases.md#verification-service

File: src/verification_service/service.rs (hunk 1 above)
Comment (tiainen) [reply src/verification_service/service.rs:102]: That will probably be enough, if the verification service only ever needs to handle payloads that are transparency logs.

Comment (tiainen) [reply src/verification_service/service.rs]: Changed the function signature to accept a slice instead of the blockchain transaction.

**Review by fishseabowl — APPROVED** [review]
Thanks for the update

**Review by johanvos — APPROVED** [review]

**Pull request merged**